test/test_autograd.py

```diff
@@ -2486,6 +2486,11 @@ def random_fullrank_matrix_distinct_singular_value(l):
     s = torch.arange(1., l + 1).mul_(1.0 / (l + 1))
     return u.mm(torch.diag(s)).mm(v.t())
 
+def random_matrix_large_singular_value(m, n):
```

Comment:
This is different from what I had in mind. It was totally my bad for not describing my proposal clearly. The tested function I was thinking is
```
def fn(x):
  S = torch.cat([x, zeros], 0)
  M = U @ S.diag() @ V
  return M.pinv()
```
and the gradcheck is run on this entire `fn` with `input` being a large positive vector (e.g., `torch.rand(...) + 1`) so that no matter how it touches `x`, as long as the perturbation is small, we always get rank `|x|`.

This might work too if the random numbers treat us well, but point perturbation touches the returned matrix, and may change its rank. Let's see what CI says.

Reply:
No problem. Will work on it, and push a commit. Thanks for clarifying.

Reply:
I think the check will work for full rank matrices, since full rank matrices are dense in the space of all matrices. This could mean that a small perturbation to the actual matrix, could with high probability land at another full rank matrix, thus causing no change in rank.